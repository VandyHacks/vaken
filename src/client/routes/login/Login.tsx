import React, { FunctionComponent } from 'react';
import { Route, Switch } from 'react-router-dom';
import STRINGS from '../../assets/strings.json';
import bg from '../../assets/img/login_bg.jpg';
import PasswordLogin from './PasswordLogin';
import OAuthLogin from './OAuthLogin';
import CreateLogin from './CreateLogin';
import Background from '../../components/Containers/Background';
import FloatingPopup from '../../components/Containers/FloatingPopup';
import Title from '../../components/Text/Title';

const LoginPage: FunctionComponent = (): JSX.Element => (
	<>
		<Background img={bg}>
			<FloatingPopup margin="1.5rem" height="25" width="27rem" backgroundOpacity="0.9">
				<Title color={STRINGS.DARK_TEXT_COLOR}>{STRINGS.FULL_NAME}</Title>
				<Switch>
					<Route path="/login/password" component={PasswordLogin} />
					<Route path="/login/create" component={CreateLogin} />
					<Route component={OAuthLogin} />
				</Switch>
			</FloatingPopup>
		</Background>
	</>
);

export default LoginPage;

// Copyright (c) 2019 Vanderbilt University
