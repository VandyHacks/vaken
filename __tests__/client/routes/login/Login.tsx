import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import Login from '../../../../src/client/routes/login/Login';

it('Login renders properly', async () => {
	const component = renderer
		.create(
			<MemoryRouter>
				<Login />
			</MemoryRouter>
		)
		.toJSON();

	expect(component).toMatchSnapshot();
});
