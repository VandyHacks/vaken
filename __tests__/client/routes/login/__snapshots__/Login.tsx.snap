// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders properly 1`] = `
<div
  className="sc-kAzzGY bLVWJd"
>
  <div
    className="sc-bdVaJa sc-bwzfXH sc-ifAKCX sc-chPdSV fTRmMz"
    height="25"
    width="27rem"
  >
    <h1
      className="sc-kgoBCf boOpPq"
      color="#3f3356"
    >
      VandyHacks
    </h1>
    <a
      href="/api/auth/google"
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <button
        className="sc-gqjmRU iicoRd"
      >
        <div
          className="sc-bdVaJa sc-bwzfXH sc-htpNat dyMbAR"
          height="min-content"
          width="min-content"
        >
          <img
            alt="Google logo"
            className="sc-iwsKbI kXBzef"
            src={[Function]}
          />
        </div>
        <div
          className="sc-htoDjs gdgsvs"
          color="#3f3356"
        >
          Sign in with Google
        </div>
      </button>
    </a>
    <a
      href="/api/auth/github"
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <button
        className="sc-gqjmRU iicoRd"
      >
        <div
          className="sc-bdVaJa sc-bwzfXH sc-htpNat dyMbAR"
          height="min-content"
          width="min-content"
        >
          <img
            alt="GitHub logo"
            className="sc-iwsKbI kXBzef"
            src={[Function]}
          />
        </div>
        <div
          className="sc-htoDjs gdgsvs"
          color="#3f3356"
        >
          Sign in with GitHub
        </div>
      </button>
    </a>
    <a
      href="/login/password"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <button
        className="sc-gqjmRU sc-fjdhpX eSCzku"
        color="white"
        fontSize="1.4em"
      >
        <div
          className="sc-htoDjs kSfDbb"
          color="white"
          fontSize="1.4em"
        >
          Sign in with email
        </div>
      </button>
    </a>
  </div>
</div>
`;
